bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,324 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.souliss.internal.handler;
+
+import java.math.BigDecimal;
+import java.net.DatagramSocket;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.souliss.SoulissBindingConstants;
+import org.openhab.binding.souliss.SoulissBindingUDPConstants;
+import org.openhab.binding.souliss.internal.SoulissDatagramSocketFactory;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingNetworkParameters;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingSendDispatcherJob;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingUDPServerJob;
+import org.openhab.binding.souliss.internal.protocol.SoulissCommonCommands;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SoulissGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Tonino Fazio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+@NonNullByDefault
+public class SoulissGatewayHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SoulissGatewayHandler.class);
+    public @Nullable DatagramSocket datagramSocketDefaultPort;
+    private @Nullable SoulissBindingUDPServerJob udpServerDefaultPortRunnableClass;
+    private SoulissCommonCommands soulissCommands = new SoulissCommonCommands();
+
+    boolean bGatewayDetected = false;
+
+    private @NonNullByDefault({}) Configuration gwConfigurationMap;
```

Comment:
You could initialize the field with an empty configuration. Then, you could remove the null annotation.

Reply:
fixed